azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -101,6 +102,19 @@ def __init__(self, auth_provider):
                 "source": ["disconnected", "disconnecting"],
                 "dest": None,
             },
+            {
+                "trigger": "_trig_enable_receive",
+                "source": "connected",
+                "dest": None,
+                "after": "_call_subscribe",
+            },
+            {"trigger": "_trig_enable_receive", "source": "connecting", "dest": None},
+            {
+                "trigger": "_trig_enable_receive",
+                "source": "disconnected",
+                "dest": "connecting",
+                "after": "_call_provider_connect",
+            },
```

Comment:
I agree that we need to discuss this because you're only fully handling one case (enable while connected) and partially handling another (if you try to enable when connected, it will connect but not subscribe).  I have some strong opinions here and I think we can make this simple and robust.

Reply:
i thought at one point to have a method that will do connect and subscribe and should be called when we are not in the connected state..i would like to know your thoughts in regards to this state machine in general..

---
In reply to: [246545470](https://github.com/Azure/azure-iot-sdk-python-preview/pull/36#discussion_r246545470) [](ancestors = 246545470)